### DISCUSSION TITLE
Identifying the object in a sentence with a compliment

### DISCUSSION THREAD

**Discussion by johnosbb** [explosion/spaCy#13124] [Help: Other Questions] [answered]
Hi,
 I am using SpaCy (nlp = spacy.load("en_core_web_sm") to try to identify the object in the following sentence.

"She considers him a friend."

Spacy tags this as follows:

```md
| Text         | Index  | POS      | Tag      | Dep      | Dep Detail               | Ancestors            | Children   | Token Head  
| ------ | ------ | ---- | ------- | ------- | --------- |  ------- | ------- | ------- |
| She          | 0      | PRON     | PRP      | nsubj    | nominal subject          | considers            |            |  considers    |
| considers    | 1      | VERB     | VBZ      | ROOT     | root                     |                      | She friend . |  considers    |
| him          | 2      | PRON     | PRP      | nsubj    | nominal subject          | friend considers     |            |  friend       |
| a            | 3      | DET      | DT       | det      | determiner               | friend considers     |            |  friend       |
| friend       | 4      | NOUN     | NN       | ccomp    | clausal complement       | considers            | him a      |  considers    |
| .            | 5      | PUNCT    | .        | punct    | punctuation              | considers            |            |  considers    |
```

"him" is identified as a nominal subject. Shouldn't it be a direct object?

**Comment by adrianeboyd** [comment, marked as answer]
I think the answer is: no, not necessarily. If you think of the bracketing as:

```
I consider [him a friend]
```

which is also similar to:

```
I consider [him to be a friend]
```

you can see how "him" could be seen as a type of subject.

This quickly gets into syntactic theory weeds, but a starting point is to look for more information on "small clauses".

If you look at p. 252 of the PTB guidelines, there's a similar example with "consider" that shows how they decided to annotate these kind of clauses:

https://www.ldc.upenn.edu/sites/www.ldc.upenn.edu/files/penn-etb-2-style-guidelines.pdf

The OntoNotes data used in `en_core_web_*` is converted from a very similar annotation scheme, so I think this is the intended analysis and not a mistake made by the parser.

However these cases are tricky, both to annotate and to parse, so expect to see some mistakes/inconsistencies in the parser output in actual use.

**Reply by johnosbb** [reply]
Thanks Adriane,

That was very helpful and the document you referenced is a great resource; I had not come across it previously.

**Comment by Ali-MH-Mansour** [comment]
Thanks for the question and the answer:
I would like to ask about a case where the parser incorrectly identifies the root.
A company **specializing** in content creation
In this structure, the root is always the subject, not the verb. 
Why?
And another question please:
Can we fix sepecific sentences' structure types? and how?
Thanks

**Reply by adrianeboyd** [reply]
In general this kind of question would be better off as a new/separate discussion thread.

A finite verb would be the root of the sentence, but here "specializing ..." is modifying the noun "company" and since there's only a noun phrase, the noun is the root. This example is similar to:

```
[A company [which is specializing in content creation]]
```

If you had "this company specializes in content creation", then "specializes" would be the root.

You can post-correct or convert dependency trees, but it can require a lot of effort/expertise to do well. I probably wouldn't recommend doing anything beyond minor edits within a spacy `Doc`, since it's not a particularly good data structure to use for editing.

I haven't used any myself, but I'm aware of some projects for converting/modifying dependency trees, e.g.:

https://gucorpling.org/depedit/

**Reply by Ali-MH-Mansour** [reply]
Thank you very much @adrianeboyd for the quick response.
Sorry for starting the discussion here, I'm new here.  In fact, I asked the question elsewhere as well.
I will keep that in mind next time.

**Reply by Ali-MH-Mansour** [reply]
I agree with you completely. But people write like this and my task is to get the verb as the root of the sentence.
Another question I asked elsewhere:
in sentences: "We help you create content". Why is "you" defined as the "subject" and not the object of the verb "help"? 
Is this considered a mistake in Spacy's training, or is there a rule behind this?

**Reply by adrianeboyd** [reply]
Instead of trying to modify the parses, I would suggest reframing your task so that you're trying to get the information you need out of the current parses.

**Comment by Ali-MH-Mansour** [comment]
I agree with you. But it is a wrong parsing or behind it you have a rule.
Just for information.

**Reply by adrianeboyd** [reply]
The parse of "We help you create content" with "you" as the subject of "create" is the intended parse based on the PTB/OntoNotes/ClearNLP annotation schemes.

**Comment by Ali-MH-Mansour** [comment]
Great
thank you very much for your response, it is very helpful.
